pkg/systemaccount/systemaccount.go

```diff
@@ -146,3 +149,18 @@ func (s *Manager) GetOrCreateProjectSystemToken(projectName string) (string, err
 	}
 	return s.userManager.EnsureToken(projectName+"-pipeline", "Pipeline token for project "+projectName, user.Name)
 }
+
+func (s *Manager) RemoveSystemAccount(userID string) error {
+	u, err := s.userManager.GetUserByPrincipalID(fmt.Sprintf("system://%s", userID))
+	if err != nil {
```

Comment:
we should not return if err is notFound

Reply:
We don't, GetUserByPrincipalID will return user as nil if not found with no error. The line below this checks if user is nil and returns nil